scripts/repo-variable-writer.go

```diff
@@ -0,0 +1,373 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"os"
+	"strings"
+	"sync"
+	"time"
+)
+
+type Project struct {
+	Name          string   `json:"name"`
+	Organization  string   `json:"organization"`
+	SubComponents []string `json:"sub_components"`
+}
+
+type Projects struct {
+	Projects []Project `json:"projects"`
+}
+
+const (
+	LocProjMeta         string = "../projects/project.json"
+	EnvGithubPatVarName string = "GH_TOKEN"
+)
+
+func toPtr[T any](v T) *T {
+	return &v
+}
+
+func handleHTTPCall(method string, url string, timeOut time.Duration,
+	body io.Reader, headers map[string]string) (*http.Response, error) {
+
+	client := &http.Client{
+		Timeout: timeOut,
+	}
+
+	req, err := http.NewRequest(method, url, body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create newRequest, method: %s, url: %s, Reason: %v", method, url, err)
+	}
+
+	for k, v := range headers {
+		req.Header.Add(k, v)
+	}
+
```

Comment:
we may want to add a debug step here.

Reply:
You mean slog.Debug()??

Reply:
I think we shouldn't attach debug log here as it will contain the Github PAT TOken